src/common/regions.ts

```diff
@@ -0,0 +1,146 @@
+import { chain, fromPairs } from 'lodash';
+import urlJoin from 'url-join';
+import US_STATE_DATASET from 'components/MapSelectors/datasets/us_states_dataset_01_02_2020.json';
+import countyAdjacencyMsa from './data/county_adjacency_msa.json';
+const { state_dataset, state_county_map_dataset } = US_STATE_DATASET;
+
+export type FipsCode = string;
+
+export enum RegionType {
+  COUNTY = 'county',
+  STATE = 'state',
+  CBSA = 'CBSA',
+}
+
+abstract class Region {
+  constructor(
+    public name: string,
+    protected urlSegment: string,
+    public fipsCode: FipsCode,
+    public population: number,
+    public regionType: RegionType,
+  ) {}
+
+  abstract fullName(): string;
+  abstract relativeUrl(): string;
+  canonicalUrl() {
+    return urlJoin('https://covidactnow.org', this.relativeUrl());
```

Comment:
Do we have some sort of environment variables for this instead of hardcoding the url?

Reply:
FWIW- We have a common/urls.ts file where I have been haphazardly trying to unify url handling...  Note that we're currently using `window.location.origin` instead of hardcoding the base URL at least in some cases.  This allows it to work on vercel preview links, etc.
https://github.com/covid-projections/covid-projections/blob/6849cf1245480ab97979ba93be2f3fd732047f36/src/common/urls.ts#L41

Also, FWIW- we do have infrastructure set up to detect whether we're on prod/staging/dev via 
https://github.com/covid-projections/covid-projections/blob/6849cf1245480ab97979ba93be2f3fd732047f36/src/common/utils/environment.ts#L7

We could use that to decide what URL to use, but as alluded to, that won't work for vercel preview links...